tests/address_test.go

```diff
@@ -1,132 +1,100 @@
-	)
-	require.NoError(err)
-	assert.NotEmpty(address.ID)
-	require.NotNil(address.Verifications)
-	require.NotNil(address.Verifications.Delivery)
-	require.GreaterOrEqual(len(address.Verifications.Delivery.Errors), 2)
-	assert.Equal(
-		"Address not found", address.Verifications.Delivery.Errors[0].Message,
-	)
-	assert.Equal(
-		"House number is missing",
-		address.Verifications.Delivery.Errors[1].Message,
+	assert := c.Assert()
+
+	address, _ := client.CreateAddress(
+		c.fixture.BasicAddress(),
+		nil,
 	)
+
+	retrievedAddress, _ := client.GetAddress(address.ID)
+
+	assert.Equal(reflect.TypeOf(&easypost.Address{}), reflect.TypeOf(retrievedAddress))
+	assert.Equal(address, retrievedAddress)
 }
 
-func (c *ClientTests) TestAddressCreationWithVerifyStrictFailure() {
+func (c *ClientTests) TestAddressAll() {
 	client := c.TestClient()
-	assert, require := c.Assert(), c.Require()
-	// Create an address with a verify strict parameter to test that it fails
-	// elegantly.
-	address, err := client.CreateAddress(
-		&easypost.Address{
-			Street1: "UNDELIEVRABLE ST",
-			City:    "San Francisco",
-			State:   "CA",
-			Zip:     "94105",
-			Country: "US",
-			Company: "EasyPost",
-			Phone:   "415-555-1212",
+	assert := c.Assert()
+
+	addresses, _ := client.ListAddresses(
+		&easypost.ListOptions{
+			PageSize: c.fixture.pageSize(),
 		},
-		&easypost.CreateAddressOptions{VerifyStrict: []string{"delivery"}},
 	)
-	assert.Nil(address)
-	require.Error(err)
-	if assert.IsType((*easypost.APIError)(nil), err) {
-		err := err.(*easypost.APIError)
-		assert.Equal("ADDRESS.VERIFY.FAILURE", err.Code)
-		assert.Equal("Unable to verify address.", err.Message)
-		require.GreaterOrEqual(len(err.Errors), 2)
-		assert.Equal("Address not found", err.Errors[0].Message)
-		assert.Equal("House number is missing", err.Errors[1].Message)
+
+	assert.LessOrEqual(len(addresses.Addresses), c.fixture.pageSize())
+	assert.NotNil(addresses.HasMore)
+	for _, address := range addresses.Addresses {
+		assert.Equal(reflect.TypeOf(&easypost.Address{}), reflect.TypeOf(address))
 	}
 }
 
-func (c *ClientTests) TestAddressUnicode() {
+func (c *ClientTests) TestAddressCreateVerify() {
+	client := c.TestClient()
+	assert := c.Assert()
+
+	address, _ := client.CreateAddress(
+		c.fixture.IncorrectAddressToVerify(),
+		&easypost.CreateAddressOptions{
+			Verify: []string{"delivery"},
+		},
+	)
+
+	assert.Equal(reflect.TypeOf(&easypost.Address{}), reflect.TypeOf(address))
+	assert.True(strings.HasPrefix(address.ID, "adr_"))
+	assert.Equal("417 MONTGOMERY ST STE 500", address.Street1)
+}
+
+func (c *ClientTests) TestAddressVerify() {
 	client := c.TestClient()
-	assert, require := c.Assert(), c.Require()
-	// Create an address with unicode field and assert if it was created
-	// correctly.
-	address, err := client.CreateAddress(
-		&easypost.Address{State: "DELEGACIóN BENITO JUáREZ"}, nil,
+	assert := c.Assert()
+
+	address, _ := client.CreateAddress(
+		c.fixture.BasicAddress(),
```

Comment:
This test is not correct (it's not verifying anything because the address was correct to begin with.) Please use the `incorrect address to verify` fixture here.

Reply:
I was following the PHP test suite, it depends on `testCreate` which uses `**basic_address()**`.

Reply:
I am going to make that change in GO, PHP needs a tweak in this unit test.

Reply:
Indeed it does, a sad side-effect from re-using objects in PHP. Are you able to put up a PR for PHP? I want you to get some exposure 😉